tests/sentry/integrations/jira/test_search_endpoint.py

```diff
@@ -0,0 +1,57 @@
+from __future__ import absolute_import
+
+import json
+
+from mock import patch
+
+from django.core.urlresolvers import reverse
+
+from sentry.integrations.jira import JiraIntegration
+from sentry.models import Integration
+from sentry.testutils import APITestCase
+
+
+SAMPLE_SEARCH_RESPONSE = """
+{
+  "expand": "names,schema",
+  "startAt": 0,
+  "maxResults": 50,
+  "total": 1,
+  "issues": [
+    {
+      "expand": "",
+      "id": "10001",
+      "self": "http://www.example.com/jira/rest/api/2/issue/10001",
+      "key": "HSP-1",
+      "fields": {
+        "summary": "this is a test issue summary"
+      }
+    }
+  ],
+  "warningMessages": [
+    "The value 'splat' does not exist for the field 'Foo'."
+  ]
+}
+"""
+
+
+class JiraSearchEndpointTest(APITestCase):
+    @patch.object(JiraIntegration, 'search_issues',
+                  return_value=json.loads(SAMPLE_SEARCH_RESPONSE.strip()))
+    def test_simple(self, mock_search_issues):
+        org = self.organization
+        self.login_as(self.user)
+
+        integration = Integration.objects.create(
+            provider='jira',
+            name='Example JIRA',
+        )
+        integration.add_organization(org.id)
+
+        path = reverse('sentry-extensions-jira-search', args=[org.slug, integration.id])
+
+        resp = self.client.get('%s?field=issue_id&query=test' % (path,))
+        assert resp.data == [
+            {'text': '(HSP-1) this is a test issue summary', 'id': 'HSP-1'}
+        ]
+        mock_search_issues.assert_called_with('test')
```

Comment:
I remember being told to not sure the `assert_called_once` kinds of things with mock (`from mock import Mock` and rather rely on `mock.call_count == some_number` instead. I'm not super familiar with what happens behind then scenes when you use `@patch` is it similar to `from mock import Mock` in that if you called a random function `mock_object.something_random('test')` it just returns true without a fuss or is this library completely different?

Reply:
yeah, `assert_called_once` isn't actually a mock method. but i'm using `assert_called_with`, which does exist: http://www.voidspace.org.uk/python/mock/mock.html?highlight=assert_called_with#mock.Mock.assert_called_with (also confirmed that changing `'test'` to another string causes the test to fail)